### ISSUE TITLE
Plugin is broken in 2.6

### ISSUE THREAD

**Issue by spuck** [alexwais/redmine_workflow_hidden_fields#5]
Just upgraded my Redmine from 2.5.2 to 2.6.0, and ran into errors with the changes in the PDF handling library in Redmine.

Here's what I've changed in lib/pdf.rb:

```
@@ -17,11 +17,6 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.

-require 'tcpdf'
-require 'fpdf/chinese'
-require 'fpdf/japanese'
-require 'fpdf/korean'

 if RUBY_VERSION < '1.9'
   require 'iconv'
 end
@@ -33,7 +28,7 @@
       include ActionView::Helpers::NumberHelper
       include IssuesHelper

-      class ITCPDF < TCPDF
+      class ITCPDF < RBPDF
         include Redmine::I18n
         attr_accessor :footer_date

```

Not sure if this fixes everything, but the migration works without error, and the workflow settings seem to be OK for now...

**Comment by JDStar** [comment]
Hi.
Me too not working in 2.6.
Install with patch @spuck without errors.
In Administration -> Workflow permission in lists option "hidden" is vivisble but after saving not remember choosen option.
![image](https://cloud.githubusercontent.com/assets/614209/5784047/edb86714-9dc8-11e4-988f-6ba073eb8481.png)

**Comment by SimonK99** [comment]
@JDStar I think i fixed the Error.

Here's what I've changed in lib\redmine_workflow_hidden_fields\workflows_helper_patch.rb:

<pre>
Line 14:
- def field_permission_tag_with_hidden(permissions, status, field, role)
+ def field_permission_tag_with_hidden(permissions, status, field, roles)

Line 20 (After Line 20-28)
-selected = permissions[status.id][name]

+if perm = permissions[status.id][name]
+       	   if perm.uniq.size > 1 || perm.size < @roles.size * @trackers.size
+         	options << [l(:label_no_change_option), "no_change"]
+         	selected = 'no_change'
+       	   else
+         	selected = perm.first
+      	   end
+     	end

Line 29/37:
- select_tag("permissions[#{name}][#{status.id}]", options_for_select(options, selected), html_options)
+ select_tag("permissions[#{status.id}][#{name}]", options_for_select(options, selected), html_options)
</pre>

**Comment by SimonK99** [comment]
@JDStar I think i fixed the Error.

Here's what I've changed in lib\redmine_workflow_hidden_fields\workflows_helper_patch.rb:




    Line 14:
    - def field_permission_tag_with_hidden(permissions, status, field, role)
    + def field_permission_tag_with_hidden(permissions, status, field, roles)
    
    Line 20 (After Line 20-28)
    -selected = permissions[status.id][name]
    +if perm = permissions[status.id][name]
    +       	   if perm.uniq.size > 1 || perm.size < @roles.size * @trackers.size
    +         	options << [l(:label_no_change_option), "no_change"]
    +         	selected = 'no_change'
    +       	   else
    +         	selected = perm.first
    +      	   end
    +     	end
    
    Line 29/37:
    - select_tag("permissions[#{name}][#{status.id}]", options_for_select(options, selected),     html_options)
    + select_tag("permissions[#{status.id}][#{name}]", options_for_select(options, selected), html_options)

**Comment by JDStar** [comment]
Thanks!
With this patch working OK.

**Comment by alexwais** [comment]
thanks, should be fixed in v0.1.3
